chain/network/src/routing.rs

```diff
@@ -497,45 +497,51 @@ impl RoutingTable {
         }
     }
 
-    /// Add this edge to the current view of the network.
-    /// This edge is assumed to be valid at this point.
-    /// Edge contains about being added or removed (this can trigger both types of events).
-    /// Return true if the edge contains new information about the network. Old if this information
-    /// is outdated.
-    pub fn process_edge(&mut self, edge: Edge) -> ProcessEdgeResult {
-        let key = edge.get_pair();
+    /// Add several edges to the current view of the network.
+    /// These edges are assumed to be valid at this point.
+    /// Return true if some of the edges contains new information to the network.
+    pub fn process_edges(&mut self, edges: Vec<Edge>) -> ProcessEdgeResult {
+        let mut new_edge = false;
+        let total = edges.len();
+
+        for edge in edges {
+            let key = edge.get_pair();
 
-        self.touch(&key.0);
-        self.touch(&key.1);
+            self.touch(&key.0);
+            self.touch(&key.1);
 
-        if !self.add_edge(edge) {
-            debug!(target:"network", "Received outdated edge.");
-            return ProcessEdgeResult { new_edge: false, schedule_computation: None };
+            if self.add_edge(edge) {
+                new_edge = true;
+            }
         }
 
-        // Minimum between known routes and 1000
-        let known_routes = std::cmp::min(self.peer_forwarding.len() as u64, 1000);
+        let mut new_schedule = None;
 
-        let new_schedule = self.recalculation_scheduled.map_or_else(
-            move || Some(Duration::from_millis(known_routes)),
-            |target| {
-                if Instant::now() > target {
-                    Some(Duration::from_millis(known_routes))
-                } else {
-                    None
-                }
-            },
-        );
+        if new_edge {
+            // Minimum between known routes and 1000
```

Comment:
how is 1000 chosen? can we extract it to a constant?

Reply:
It is the maximum time we will wait before updating the routing table. We don't want to be out of sync for too long. Right now it 1000ms (1s).